Split display widgets documentation

guillempages opened the pull request:
## Description:

The display component has many sub-components to be displayed in it (graphs, images, animations,...).
In order to improve maintainability, search and findability, splitting them into their own "full fledged" components.

**Related issue (if applicable):** NA

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** NA
## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

guillempages committed:
Split display components into own pages

The display component has many sub-components to be displayed in it (graphs, images, animations,...).
In order to improve maintainability, search and findability, splitting them into their own "full fledged" components.

guillempages commented on components/display/images.rst:
```diff
@@ -0,0 +1,122 @@
+.. code-block:: yaml
+
+    image:
+      - file: https://esphome.io/_images/logo.png
```
No; this one is an example to be copy&pasted somewhere else, and thus requires the full URL

nagyrobi commented:
I would have done this after finishing LVGL and merging it.

guillempages commented:
lint is complaining about an absolute link to https://esphome.io. In this particular case, the link is part of a code example, and IMHO it must stay as an absolute link; otherwise copy&pasting of the example would not work.

guillempages commented:
> I would have done this after finishing LVGL and merging it.

Do you mean I should close the PR and we should wait, or that I just took you some work?

guillempages commented on components/display/images.rst:
```diff
@@ -0,0 +1,122 @@
+.. code-block:: yaml
+
+    image:
+      - file: https://esphome.io/_images/logo.png
```
This URL is part of an example, and as such must stay absolute; even if it contains esphome.io

nagyrobi commented:
> lint is complaining about an absolute link to https://esphome.io. In this particular case, the link is part of a code example, and IMHO it must stay as an absolute link; otherwise copy&pasting of the example would not work.

There's a config file near the index where exceptions can be added.

nagyrobi commented:
> > I would have done this after finishing LVGL and merging it.
> 
> Do you mean I should close the PR and we should wait, or that I just took you some work?

I'd say let's set it to draft, now that you've put the effort in. Will finish together.

guillempages commented:
> > lint is complaining about an absolute link to https://esphome.io. In this particular case, the link is part of a code example, and IMHO it must stay as an absolute link; otherwise copy&pasting of the example would not work.
> 
> There's a config file near the index where exceptions can be added.

I think I found it; it's the lint.py script itself. At least there was already an exception there for the `components/display/index.rst` file; which originally had the problematic example. Changed this to point to the new images.rst files instead.

jesserockz commented on components/display/animation.rst:
While these are all "display" related components, they should be directly in the `components` directory, the `display` directory is for display platforms only.

guillempages replied:
done

guillempages committed:
Move display components files one level up

Though they are very closely related to display; they are components in their own right, and thus should be moved one level up.

guillempages committed:
Fix lint issue

nagyrobi committed:
Merge branch 'current' into split_display_widgets

guillempages committed:
Move missing image

jesserockz commented on components/images.rst:
```diff
@@ -0,0 +1,122 @@
+
+    To use images you will need to have the python ``pillow`` package installed.
+    If you're running this as a Home Assistant add-on or with the official ESPHome docker image, it should already be
+    installed. Otherwise you need to install it using ``pip install pillow``.
+    Additionally, if you want to use SVG images (including MDI images), you will additionally need to have the python ``cairosvg`` package installed.
+    If you're running this as a Home Assistant add-on or with the official ESPHome docker image, it should also already be
+    installed. Otherwise you need to install it using ``pip install cairosvg``.
```
```suggestion
    installed.
    Additionally, if you want to use SVG images (including MDI images), you will additionally need to have the python ``cairosvg`` package installed.
    
    Use ``pip install "esphome[displays]"`` to install these optional dependencies with
    the versions that ESPHome requires.
```

jesserockz committed:
Update components/images.rst

jesserockz commented on components/images.rst:
```diff
@@ -0,0 +1,123 @@
+
+.. note::
+
+    To use images you will need to have the python ``pillow`` package installed.
+    If you're running this as a Home Assistant add-on or with the official ESPHome docker image, it should already be
+    installed.
+    Additionally, if you want to use SVG images (including MDI images), you will additionally need to have the python ``cairosvg`` package installed.
+    
+    Use ``pip install "esphome[displays]"`` to install these optional dependencies with
+    the versions that ESPHome requires.
```
```suggestion
    To use images you will need to have the python ``pillow`` package installed.
    Additionally, if you want to use SVG images (including MDI images), you will
    additionally need to have the python ``cairosvg`` package installed.

    If you're running this as a Home Assistant add-on or with the official ESPHome docker image, it should already be installed.

    Use ``pip install "esphome[displays]"`` to install these optional dependencies with
    the versions that ESPHome requires.
```

jesserockz committed:
Update components/images.rst
